server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -260,6 +272,11 @@ private static IllegalArgumentException aliasesNotSupportedException(String expr
                 "alias, specify the corresponding concrete indices instead.");
     }
 
+    private static IllegalArgumentException dataStreamsNotSupportedException(String expression) {
+        return new IllegalArgumentException("The provided expression [" + expression + "] matches a " +
+            "data stream, specify the corresponding concrete indices instead.");
```

Comment:
Should the error message be updated to suggest specifying either concrete indices or aliases since either are legally resolvable if only `IGNORE_DATA_STREAMS` is specified?